### ISSUE TITLE
Generated pom may resolve the wrong version of a dependency if a version property in an imported bom has been overridden

### ISSUE THREAD

**Issue by wilkinsona** [spring-gradle-plugins/dependency-management-plugin#184]
When the plugin configures a project's pom, it adds imports to the `<dependencyManagement>` section for any boms that have been imported. If a version property in an imported bom has been overridden, this won't be reflected in the generated pom and the wrong versions may be resolved by Maven.

Unfortunately, Maven doesn't support overriding version properties when a bom is imported (it's only supported when a bom's used as a parent) so fixing this isn't as simple as copying over the property overrides. I think what needs to happen is:

1. Add the bom import to the generated pom
2. Identify any dependencies in the bom with overridden versions
3. Add explicit dependency management for those dependencies to the generated pom using the overridden version

**Comment by wilkinsona** [comment]
@dsyer Things should be better now in 1.0.4.BUILD-SNAPSHOT available from https://repo.spring.io/plugins-snapshot. Can you please try it out and let me know how you get on. If it fixes your problem, I'll get on with 1.0.4.RELEASE.

**Comment by dsyer** [comment]
Works for me now, thanks.


```
buildscript {
	ext {
		springBootVersion = '1.5.10.BUILD-SNAPSHOT'
		wrapperVersion = '1.0.9.BUILD-SNAPSHOT'
	}
	repositories {
		mavenLocal()
		mavenCentral()
		maven { url "https://repo.spring.io/plugins-snapshot" }
		maven { url "https://repo.spring.io/plugins-milestone" }
	}
	dependencies {
		classpath("org.springframework.boot.experimental:spring-boot-thin-gradle-plugin:${wrapperVersion}")
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
		// Remove this when Boot 1.5.10 is out.
		classpath "io.spring.gradle:dependency-management-plugin:1.0.4.BUILD-SNAPSHOT"
	}
}
```

Maybe an example in that style in the README would be a good idea (what to do to use with Spring Boot)?
